### ISSUE TITLE
[sysrst_ctrl] D3 Review Opens

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#13995]
- [x] Rerun with Xcelium and promote array connection error in MUBI syncs to an error @sriyerg. split out as a separate issue: #14032
- [x] Open an issue for making the signal names more generic (track as a future item). #14033
- [x] Fix the the IRQ clearing priority in RTL. #14042
- [x] Nit: rst_req_o doesn’t really need the `aon_sysrst_ctrl` prefix, similar to wkup_req_o #14042
- [x] Nit: It feels like intr_sysrst_ctrl_o should have a slightly different name, since the `sysrst_ctrl` portion is already implied by top level connection. Ideally it would just be `intr_o`, but I don’t think our top level integration supports an empty string as a name (adc_ctrl had the same issue). #14042
- [x] Nit: There are a few cases of counters where cnt_d is constructed separately and just assigned to cnt_q. Some older synthesis tools don’t power compile this very well, since they don’t understand how to do a cnt_d^cnt_q to determine an update should be done. I believe this is the case with our project also.  This is not a must fix, but something to check with Nuvoton if the two would actually behave differently.  There’s a chance the enable from the construction of the cnt_d will propagate through already, and tools are smart enough to handle it. #14042
- [x] A few detectors have configurable debounce values but a 0 detect value. Looking at the RTL, this seems to suggest it would just end up sampling 3 times, is this what’s desired? detect once -> wait debounce time -> detect again -> immediately detect again.

**Comment by msfschaffner** [comment]
@tjaychen for the question about detector behavior, is your concern that we do not continuously sample the signal while debouncing?

if so, we may already have discussed this here:

https://github.com/lowRISC/opentitan/issues/10663
https://github.com/lowRISC/opentitan/pull/10636

TL;DR; we switched back to this behavior (which was originally implemented by Eric) as it easier to test and interact with from the software / DV side.

**Comment by msfschaffner** [comment]
Regarding power compile: I ran a couple of experiments, and it appears that DC is smart enough to figure out how to clock gate counters formulated in this manner. 

However, one change that we should make is remove the constant assertion of `cnt_clr` in the Idle state of the debouncer FSM. I.e. if we always clear and otherwise count, there are effectively no cycles where the flop is NOT enabled, and hence no clock gate is inferred.

I made the change as part of #14042.

**Comment by tjaychen** [comment]
> @tjaychen for the question about detector behavior, is your concern that we do not continuously sample the signal while debouncing?
> 
> if so, we may already have discussed this here:
> 
> #10663 #10636
> 
> TL;DR; we switched back to this behavior (which was originally implemented by Eric) as it easier to test and interact with from the software / DV side.

sounds good, thanks for looking it up.

**Comment by tjaychen** [comment]
> Regarding power compile: I ran a couple of experiments, and it appears that DC is smart enough to figure out how to clock gate counters formulated in this manner.
> 
> However, one change that we should make is remove the constant assertion of `cnt_clr` in the Idle state of the debouncer FSM. I.e. if we always clear and otherwise count, there are effectively no cycles where the flop is NOT enabled, and hence no clock gate is inferred.
> 
> I made the change as part of #14042 and took note that we should double check with Nuvoton also.

sounds great. Do you actually want to add this as part of style guidance? Otherwise people like me might forget and ask again in a few months.

**Comment by msfschaffner** [comment]
> > Regarding power compile: I ran a couple of experiments, and it appears that DC is smart enough to figure out how to clock gate counters formulated in this manner.
> > However, one change that we should make is remove the constant assertion of `cnt_clr` in the Idle state of the debouncer FSM. I.e. if we always clear and otherwise count, there are effectively no cycles where the flop is NOT enabled, and hence no clock gate is inferred.
> > I made the change as part of #14042 and took note that we should double check with Nuvoton also.
> 
> sounds great. Do you actually want to add this as part of style guidance? Otherwise people like me might forget and ask again in a few months.

Yes, good idea - let's add this to the style guide.
